hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/HandleContextImpl.java

```diff
@@ -405,8 +459,17 @@ private void dispatchSyntheticTxn(
         }
 
         final var childStack = new SavepointStackImpl(current().state(), configuration());
+        HederaFunctionality function;
+        try {
+            function = functionOf(txBody);
+        } catch (UnknownHederaFunctionality e) {
+            logger.error("Possible bug: unknown function in transaction body", e);
```

Comment:
I am surprised that this is needed. The call of `dispatcher.dispatchPureChecks()` above should already fail if the functionality was unknown.

Reply:
The problem is that `functionOf` throws a checked `UnknownHederaFunctionality`, and I didn't any existing reference to the functionality of the child txbody, so I think I had to do this.